sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/QuerySamples.cs

```diff
@@ -74,15 +76,99 @@ await foreach (TimeSeriesPoint point in humidityEventsQueryAnalyzer.GetResultsAs
+
+            var celsiusVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature"),
+                new TimeSeriesExpression("avg($value)"));
+            var fahrenheitVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature * 1.8 + 32"),
+                new TimeSeriesExpression("avg($value)"));
+
+            var celsiusVariableName = "TemperatureInCelsius";
+            var fahrenheitVariableName = "TemperatureInFahrenheit";
+            var variables = new Dictionary<string, TimeSeriesVariable>
+            {
+                { celsiusVariableName, celsiusVariable },
+                { fahrenheitVariableName, fahrenheitVariable }
+            };
+
+            timeSeriesTypes.Add(new TimeSeriesType("TemperatureSensor", variables) { Id = "TemperatureSensorTypeId" });
+
+            Response<TimeSeriesTypeOperationResult[]> createTypesResult = await client
+                .Types
+                .CreateOrReplaceAsync(timeSeriesTypes)
+                .ConfigureAwait(false);
+
+            if (createTypesResult.Value.First().Error != null)
+            {
+                Console.WriteLine($"\n\nFailed to create a Time Series Insights type. " +
+                    $"Error Message: '{createTypesResult.Value.First().Error.Message}.' " +
+                    $"Code: '{createTypesResult.Value.First().Error.Code}'.");
+            }
+            else
+            {
+                // Get the Time Series instance and replace its type with the one we just created
+                Response<InstancesOperationResult[]> getInstanceResult = await client.Instances.GetAsync(new List<TimeSeriesId> { tsId });
+                if (getInstanceResult.Value.First().Error != null)
+                {
+                    Console.WriteLine($"\n\nFailed to retrieve Time Series instance with Id '{tsId}'. " +
+                        $"Error Message: '{getInstanceResult.Value.First().Error.Message}.' " +
+                        $"Code: '{getInstanceResult.Value.First().Error.Code}'.");
+                }
+                else
+                {
+                    TimeSeriesInstance instanceToReplace = getInstanceResult.Value.First().Instance;
+                    instanceToReplace.TypeId = createTypesResult.Value.First().TimeSeriesType.Id;
+                    Response<InstancesOperationResult[]> replaceInstanceResult = await client.Instances.ReplaceAsync(new List<TimeSeriesInstance> { instanceToReplace });
+                    if (replaceInstanceResult.Value.First().Error != null)
+                    {
+                        Console.WriteLine($"\n\nFailed to retrieve Time Series instance with Id '{tsId}'. " +
+                            $"Error Message: '{replaceInstanceResult.Value.First().Error.Message}.' " +
+                            $"Code: '{replaceInstanceResult.Value.First().Error.Code}'.");
+                    }
+                    else
```

Comment:
Thinking, there are many nested else here. I do understand error handling and only in happy path we want to continue with other operations. Wondering, is this recommended per guidelines? The trade off here is readability of sample and long code.

Reply:
Good point. Let me follow the same pattern we've been following elsewhere in the sample.